core/global/input_manager.gd

```diff
@@ -147,23 +159,42 @@ func _on_gamepad_change(_device: int, _connected: bool) -> void:
 	for gamepad in managed_gamepads.values():
 		if gamepad.phys_path in discovered_paths:
 			continue
-		logger.debug("Gamepad no longer exists: " + gamepad.phys_path)
-		gamepad.virt_device.close()
+
+		logger.debug("Gamepad disconnected: " + gamepad.phys_path)
```

Comment:
If we never close the file descriptor for these orphaned devices, won't there be issues when it reconnects?

Reply:
We're re-associating the physical device with its existing virtual device. This should ensure that players don't need to swap controllers if they are disconnected for whatever reason.

Reply:
It also makes the logic easier in handheld controller, I don't need to reset the gamepad device and add checks that it exists, which allows the user to still send events with the kb device even when in mouse mode for example.

Reply:
This effectively stops players from fully disconnecting any controllers, as the virtual gamepad will always exist. It doesn't seem like it would cause issues most of the time, but I wonder if we'll see a problem with this in the future.

Reply:
> It also makes the logic easier in handheld controller, I don't need to reset the gamepad device and add checks that it exists, which allows the user to still send events with the kb device even when in mouse mode for example.

Actually, NVM the way I have it works.